presto-spi/src/main/java/com/facebook/presto/spi/security/AccessControl.java

```diff
@@ -46,7 +48,7 @@ default AuthorizedIdentity selectAuthorizedIdentity(Identity identity, AccessCon
      * Check if the query is unexpectedly modified using the credentials passed in the identity.
      * @throws com.facebook.presto.spi.security.AccessDeniedException if query is modified.
      */
-    void checkQueryIntegrity(Identity identity, AccessControlContext context, String query);
+    void checkQueryIntegrity(Identity identity, AccessControlContext context, String query, Map<QualifiedObjectName, ViewDefinition> viewTextDefinitions, Map<QualifiedObjectName, MaterializedViewDefinition> materializedViewTextDefinitions);
```

Comment:
I would call these viewDefinitions and materializedViewDefinitions. it includes the whole view definition and not just the query text.